kanidmd/lib/src/entry.rs

```diff
@@ -1795,6 +1795,24 @@ impl Entry<EntryReduced, EntryCommitted> {
                         atype: "entrydn".to_string(),
                         vals: vec![dn.as_bytes().to_vec()],
                     }),
+                    "mail;primary" | "mailprimary" | "emailprimary" => {
+                        attr_map.get(kani_a).map(|pvs| LdapPartialAttribute {
+                            atype: ldap_a.to_string(),
+                            vals: pvs
+                                .first()
+                                .map(|first| vec![first.clone()])
+                                .unwrap_or_default(),
+                        })
+                    }
+                    "mail;alternative" | "mailalternative" | "emailalternative" => {
```

Comment:
Since you only add the phantoms for `emailalternative` then you should remove `mailalternative`. I think that mail;alternative can stay though since that wouldn't be allow by iname rules

Reply:
Done.